src/stores/Object.js

```diff
@@ -709,34 +749,60 @@ ReplaceMetadata = flow(function * ({
       service: "search"
     });
 
-    const {lro_handle} = yield Fabric.CallBitcodeMethod({
-      libraryId,
-      objectId,
-      writeToken,
-      method,
-      constant,
-      service: "search"
-    });
-
-    let done;
-    let lastRunTime;
-    while(!done) {
-      let results = yield Fabric.CallBitcodeMethod({
+    let lroHandle;
+    try {
+      const response = yield Fabric.CallBitcodeMethod({
         libraryId,
         objectId,
         writeToken,
-        method: "crawl_status",
-        body: {"lro_handle": lro_handle},
-        constant: false,
+        method: "search_update",
+        constant,
         service: "search"
       });
 
+      lroHandle = response.lro_handle;
+    } catch(error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      throw error;
+    }
+
+    yield new Promise(resolve => setTimeout(resolve, 60000));
+
+    let done;
+    let lastRunTime;
+    while(!done) {
+      let results;
+
+      try {
+        results = yield Fabric.CallBitcodeMethod({
+          libraryId,
+          objectId,
+          writeToken,
+          method: "crawl_status",
+          body: {"lro_handle": lroHandle},
+          constant: false,
+          service: "search"
+        });
+      } catch(error) {
+        // eslint-disable-next-line no-console
+        console.error("Failed to retrieve crawl status", error);
+        done = true;
+      }
+
       if(results) {
         lastRunTime = results.custom.duration;
-        if(results.custom.run_state === "finished") done = true;
+
+        if(
+          results.custom.run_state === "finished" ||
+          results.custom.run_state === "failed" ||
+          results.state === "terminated"
+        ) {
+          done = true;
+        }
       }
 
-      yield new Promise(resolve => setTimeout(resolve, 1000));
+      yield new Promise(resolve => setTimeout(resolve, 30000));
```

Comment:
Does this timeout limit how long we are waiting for the crawler to finish?  Ideally we would wait until it finishes without a limit (or timeout).

Reply:
It does add time to the total crawler time. This was added based on Marc's script, which either waits 30 seconds or one minute in between crawl status checks. @elv-marc Do you remember the significance of waiting in between `crawl_status` calls?

Reply:
@elv-serban Do you want us to test updating the index without this timeout?

Reply:
There is a status API.  These operations normally take a long time (several minutes) so we don't have to check super-frequently but something like 2-5 seconds would be fine as far as polling frequency.

Reply:
Kevin, is there a better way to check every few seconds? This wait is after the status call. I didn't use `setInterval` because I wanted to use async/await

Reply:
@elv-kevin Update. I tried to use a combination of promises and setInterval to make the status call every few seconds. Still was treated as a macro task and gets fired in the wrong order. I need it to call `crawl_status` repeatedly until it finishes or fails AND THEN move on. Not sure there's any other way to do it

Reply:
Removed 60 second wait outside of loop and moved 30 second wait to the top of the loop.